x-pack/legacy/plugins/lens/public/indexpattern_plugin/state_helpers.ts

```diff
@@ -64,11 +64,20 @@ export function updateColumnParam<
 export function changeColumn(
   state: IndexPatternPrivateState,
   columnId: string,
-  newColumn: IndexPatternColumn
+  newColumn: IndexPatternColumn,
+  { keepParams }: { keepParams: boolean } = { keepParams: true }
 ) {
+  const oldColumn = state.columns[columnId];
+
   const newColumns: IndexPatternPrivateState['columns'] = {
     ...state.columns,
-    [columnId]: newColumn,
+    [columnId]:
+      keepParams &&
```

Comment:
Just a nit: I found this a bit gnarly to read. It might be better to pull the boolean clause out into a const, so that this is more obvious at a glance.